src/System/Diagnostic/SourceCodeIntegrityChecker.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\System\Diagnostic;
+
+use FilesystemIterator;
+use Glpi\Toolbox\VersionParser;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use SebastianBergmann\Diff\Differ;
+use SebastianBergmann\Diff\Output\UnifiedDiffOutputBuilder;
+
+/**
+ * @since 10.1.0
+ */
+class SourceCodeIntegrityChecker
+{
+    public const STATUS_OK = 0;
+    public const STATUS_ALTERED = 1;
+    public const STATUS_MISSING = 2;
+    public const STATUS_ADDED = 3;
+
+    /**
+     * @return array
+     */
+    public function getPathsToCheck(): array
+    {
+        return [
+            'ajax', 'css', 'front', 'inc', 'install', 'js', 'lib', 'locales', 'pics',
+            'public', 'resources', 'sound', 'src', 'templates', 'vendor',
+            'api.php', 'apirest.md', 'apirest.php', 'caldav.php', 'index.php', 'status.php'
+        ];
+    }
+
+    /**
+     * @return array
+     * @phpstan-return array{algorithm: string, files: array<string, string>}|null
+     */
+    private function getBaselineManifest(): array
```

Comment:
To prevent the following error:
```
Uncaught Exception TypeError: Glpi\System\Diagnostic\SourceCodeIntegrityChecker::getBaselineManifest(): Return value must be of type array, null returned in /var/www/glpi/src/System/Diagnostic/SourceCodeIntegrityChecker.php at line 80
```

```suggestion
    private function getBaselineManifest(): ?array
```